tfx_addons/sampler/executor.py

```diff
@@ -0,0 +1,259 @@
+# Copyright 2021 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Executor for TensorFlow Transform."""
+
+import os
+import random
+from typing import Any, Dict, List, Text
+
+import apache_beam as beam
+import tensorflow as tf
+from tfx import types
+from tfx.dsl.components.base import base_beam_executor
+from tfx.dsl.io import fileio
+from tfx.types import artifact_utils
+from tfx.utils import io_utils, json_utils
+
+from tfx_addons.sampler import spec
+
+
+class Executor(base_beam_executor.BaseBeamExecutor):
+  """Executor for Sampler."""
+  def _CreatePipeline(self, unused_transform_output: Text) -> beam.Pipeline:
+    """Creates beam pipeline.
+    Args:
+      unused_transform_output: unused.
+    Returns:
+      Beam pipeline.
+    """
+
+    return self._make_beam_pipeline()
+
+  def Do(
+      self,
+      input_dict: Dict[Text, List[types.Artifact]],
+      output_dict: Dict[Text, List[types.Artifact]],
+      exec_properties: Dict[Text, Any],
+  ) -> None:
+    """Sampler executor entrypoint.
```

Comment:
For all docstrings you have, add some additional space for the reader's benefit. Here's an example of a great docstring to follow closely: https://github.com/tensorflow/tfx/blob/master/tfx/components/example_gen/base_example_gen_executor.py#L228

Notice the indents in the args and the newlines between sections.